### PR TITLE
HolidayCalendarService SPI for dynamic holiday calendar loading

### PR DESCRIPTION
**Pull request by davejoyce-lab49**

### CODE REVIEW

**Review by davejoyce — APPROVED** [review]
OK - ready to merge

File: holiday-calendar-base/src/main/java/com/github/davejoyce/calendar/SpecialAnniversary.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    public SpecialAnniversary(String name,
+                              String description,
+                              String anniversaryDate,
+                              boolean rollable) {
+        this(name, description, LocalDate.parse(anniversaryDate), rollable);
+    }
+
+    public LocalDate getAnniversaryDate() {
+        return anniversaryDate;
+    }
```
Comment (battleblow) [inline holiday-calendar-base/src/main/java/com/github/davejoyce/calendar/SpecialAnniversary.java:56]: `@Getter`?

**Review by battleblow — APPROVED** [review]
LGTM.  Lombok may allow some simplifications if you wanted to use that.

File: holiday-calendar-base/src/main/java/com/github/davejoyce/calendar/Holiday.java [hunk 2]
```diff
@@ -165,6 +166,11 @@ public HolidayBuilder observance(final Observance observance) {
             return this;
         }
 
+        public HolidayBuilder anniversaryDate(final LocalDate anniversaryDate) {
+            this.anniversaryDate = anniversaryDate;
+            return this;
+        }
+
```
[the remark this answers is not in the document]
Comment (davejoyce) [reply holiday-calendar-base/src/main/java/com/github/davejoyce/calendar/Holiday.java:173]: @battleblow I will open a new issue to refactor to use Lombok for this. Good suggestion.

File: holiday-calendar-base/src/main/java/com/github/davejoyce/calendar/SpecialAnniversary.java (hunk 1 above)
Comment (davejoyce) [reply holiday-calendar-base/src/main/java/com/github/davejoyce/calendar/SpecialAnniversary.java:56]: @battleblow will be part of the refactor in new issue.

**Pull request merged**